CONTRIBUTING.md

```diff
@@ -1,133 +1,438 @@
-# Contributing to Apache NuttX RTOS
+# Apache NuttX RTOS Contributing Guidelines
 
-Hi! Thank you for your interest in contributing to Apache NuttX RTOS :-)
+Thank you for your interest in contributing to Apache NuttX RTOS :-)
 
-## Guidelines
+If you haven't yet read
+[The Inviolable Principles of NuttX]( https://github.com/apache/nuttx/blob/master/INVIOLABLES.md)
+please do so right now.
 
-To help us successfully review your contribution, it is very
-important that you follow these guidelines.
+Please note that NuttX supports over 15 different architectures, 360+ boards,
+and 1600+ configurations, that are used in various project and products around
+the world. Remember that any code change may affect different users and their
+business. Please try to keep your contributions high quality and compatible.
 
-If you need more information please read the [Full Contributing Documentation](https://nuttx.apache.org/docs/latest/contributing/index.html).
+To help us smoothly process your contributions it is very important that you
+follow the guidelines. These rules are based on daily experiences and helps
+us preserve long term self-compatibility and maintenance of the project.
+NuttX is maintained by a small team of volunteers from around the world.
 
-### Coding Standard
+This document is split into two main parts:
 
-* You should follow [NuttX C Coding Standard](https://nuttx.apache.org/docs/latest/contributing/coding_style.html).
+ 1. Contribution Rules.
+ 2. Contribution Templates.
 
-* Your code will be automatically checked by GitHub Continuous Integration
-  (CI) system. If you see the "check" step fails, it is possible that this
-  happens due to style errors.
+If you need more information please read the
+[Full Contributing Documentation](https://nuttx.apache.org/docs/latest/contributing/index.html)
+or ask questions at our [Social Media channels](https://nuttx.apache.org/community).
```

Comment:
Should we direct questions to the (archived) mailing list?

Reply:
I think this is shortest and most direct way to point where users may search / ask for help? Please provide text suggestions :-)

Reply:
We could link to the NuttX Community page of the project website, which has mailing lists, social media, etc., and let the reader decide which to use... https://nuttx.apache.org/community/ We can leave it as is for now...